lib/Sema/TypeCheckDecl.cpp

```diff
@@ -957,8 +957,7 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
     CanType otherSigType = other->getOverloadSignatureType();
 
     // If there is another conflict, complain.
-    if (currentSigType == otherSigType ||
-        currentSig.Name.isCompoundName() != otherSig.Name.isCompoundName()) {
+    if (conflicting(currentSig, currentSigType, otherSig, otherSigType)) {
```

Comment:
This is going to redo the check from earlier. Is it possible to not do that?

Reply:
@jrose-apple I agree it's not ideal. We could pass a `bool` flag, something like `areSignaturesConflicting` – at the time I didn't really feel it was worth it given the logic being repeated was fairly lightweight, but am happy to do so if you feel it would be worthwhile.